### DISCUSSION TITLE
big question

### DISCUSSION THREAD

**Discussion by xIneffable** [PrismarineJS/mineflayer#2231] [Q&A]
So, i I have a Russian launcher called Vimeworld (Vimeworld.ru). This launcher complements modifications for the Minecraft game. And so, this launcher requires data (login and password). After entering the Vimeworld launcher, you can go to the game server (ip: vimeworld.net:25565).
I tried to write a code to enter the game, but the program stops working and does not give any error, what may be related to this. Please note that you cannot get to the Vimeworld server through the tlauncher, ONLY THROUGH THEIR OWN LAUNCHER!
Also you should to know that without registration of your acc you can't join the server. To make the registration you can only through their website (VimeWorld.com)
My code is:
```
const mineflayer = require("mineflayer")

const bot = mineflayer.createBot({
    username: "Bot12",
    password: "mypass",
    host: "vimeworld.net",
    port: 25565,
    version: '1.8.8'
})

function greetings() {
    bot.chat('Hi, I am bot!');
}
bot.once('spawn', greetings)
```

**Comment by simonjs830** [comment]
I experience similar behavior when trying to log in with invalid credentials.

**Comment by xIneffable** [comment]
I think that i should find where goes the auth of Vimeworld like yggdrasil. But generally I don't know what to do

**Comment by xIneffable** [comment]
Also, i discovered that you can play with bots in singular mode via their launcher (cracked). But when I'm trying to play on servers it gets an error like: You can't join server, restart launcher (smth like that)

**Comment by raymondjxu** [comment]
Sounds like their launcher sends a custom packet and/or sends a custom brand. **This is not a bug with Mineflayer**. If you want to be able to connect using Mineflayer, you’ll have to reverse-engineer the packets they send and manually emit them.

**Comment by xIneffable** [comment]
How should I do that and what to use?

**Comment by WhoTho** [comment]
One thing to reverse engineer the packet information is by using [pakkit](https://github.com/Heath123/pakkit). This is a minecraft proxy server that logs all the packets going to and from the client. You can use this to see what custom packets are being sent during login and then replicate them yourself.
You would have to use 
```js
bot._client.write(“custom_packet_name”, {
flags
})
```

**Comment by xIneffable** [comment]
> One thing to reverse engineer the packet information is by using [pakkit](https://github.com/Heath123/pakkit). This is a minecraft proxy server that logs all the packets going to and from the client. You can use this to see what custom packets are being sent during login and then replicate them yourself.
> You would have to use 
> ```js
> bot._client.write(“custom_packet_name”, {
> flags
> })
> ```

Wow, thank you so much for your good answer. As a beginner you help me so much
